src/libraries/Common/src/Interop/OSX/Swift.Runtime/Foundation.cs

```diff
@@ -0,0 +1,217 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+#pragma warning disable CS3016 // Arrays as attribute arguments are not CLS Compliant
+#pragma warning disable SYSLIB1051
+
+namespace Swift.Runtime
+{
+    /// <summary>
+    /// Represents a Swift type in C#.
+    /// </summary>
+    internal unsafe interface ISwiftType
+    {
+        public abstract void* Metadata();
+    }
+
+    /// <summary>
+    /// Represents a Swift protocol in C#.
+    /// </summary>
+    internal unsafe interface ISwiftProtocol
+    {
+       public abstract void* WitnessTable(ISwiftType type);
```

Comment:
Make this a static abstract instead as we don't need netstandard2.0 support?

Then in the usage in the CryptoKit interop, we don't need to create a default instance of a protocol to get the witness table.